cpp/assetmanager.cpp

```diff
@@ -62,13 +52,14 @@ Texture *AssetManager::load_texture(const std::string &name) {
 		if (wd < 0) {
 			throw util::Error{"failed to add inotify watch for %s", filename.c_str()};
 		}
-		this->watch_fds[wd] = ret;
+		this->watch_fds[wd] = ret.get();
 #endif
 	}
 
-	this->textures[filename] = ret;
-
-	return ret;
+	// Insert the texture into the map and return the texture.
+	auto emplacedAt = textures.emplace(std::move(filename), std::move(ret)).first;
+	tex_ptr& tex = emplacedAt->second;
+	return tex.get();
```

Comment:
do you think it's a good idea to use a nice `unique_ptr` but then return a raw pointer?